src/components/Slider/useSlider.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  ARROW_DOWN,
+  ARROW_LEFT,
+  ARROW_RIGHT,
+  ARROW_UP,
+  END,
+  HOME,
+  PAGE_DOWN,
+  PAGE_UP,
+} from '@constants/key';
+import { KeyboardEvent, MouseEvent, useRef, useState } from 'react';
+
+import { SliderProps } from '.';
+
+const useSlider = ({
+  min,
+  max,
+  defaultValue,
+  size,
+  step,
+  orientation,
+}: Omit<SliderProps, 'label' | 'children'>) => {
+  const [value, setValue] = useState<number>(defaultValue);
+
+  const defaultRatio = Math.round(((defaultValue - min) / (max - min)) * 100);
+  const [thumbPosition, setThumbPosition] = useState<number>(defaultRatio);
+  const [filledRatio, setFilledRatio] = useState<number>(defaultRatio);
+
+  const trackRef = useRef<HTMLDivElement | null>(null);
+  const filledRef = useRef<HTMLDivElement | null>(null);
+  const thumbRef = useRef<HTMLDivElement | null>(null);
+
+  const isHorizontal = orientation === 'horizontal';
+
+  const getValue = () => value;
+
+  const getStyles = () => {
+    const SLIDER_THICKNESS = 4;
+
+    const rootStyle = {
+      horizontal: {
+        width: `${size}px`,
+      },
+      vertical: {
+        height: `${size}px`,
+      },
+    };
+
+    const trackStyle = {
+      horizontal: {
+        width: `${size}px`,
+        height: `${SLIDER_THICKNESS}px`,
+      },
+      vertical: {
+        height: `${size}px`,
+        width: `${SLIDER_THICKNESS}px`,
+      },
+    };
+
+    const filledStyle = {
+      horizontal: {
+        width: `${filledRatio}%`,
+        height: `${SLIDER_THICKNESS}px`,
+      },
+      vertical: {
+        height: `${filledRatio}%`,
+        width: `${SLIDER_THICKNESS}px`,
+      },
+    };
+
+    const thumbStyle = {
+      horizontal: {
+        left: `${thumbPosition}%`,
+        transform: 'translateX(-50%)',
+      },
+      vertical: {
+        bottom: `${thumbPosition}%`,
+        transform: 'translateY(50%)',
+      },
+    };
+
+    return {
+      rootStyle: rootStyle[orientation],
+      trackStyle: trackStyle[orientation],
+      filledStyle: filledStyle[orientation],
+      thumbStyle: thumbStyle[orientation],
+    };
+  };
+
+  const onMoveSlider = (
+    e: MouseEvent<HTMLDivElement> | globalThis.MouseEvent,
```

Comment:
헉 globalThis 이건 뭐죠? 처음봐요

Reply:
브라우저에서는 window가 될텐데 없어도 되는 네임스페이스라고 생각해요 ~

Reply:
이거는 내일 회의 때 공유할게요

Reply:
지금보니 ... 타입명이 중복이네요 ... 근데 저는 globalThis보다 window로 쓰는게 더 좋을 것 같아요. 근데 안주무시나요?

Reply:
```suggestion
    e: ReactMouseEvent<HTMLDivElement> | MouseEvent,
```

이렇게 변경하겠습니다 감사합니다~

<br />

**_jsx에 붙일 때는 react 이벤트로 타이핑이 되어 있어야 하고_** 
_**DOM에 직접 붙일 때는 native 이벤트여야 한다**_
_**- se030, prayinforrain -**_